### ISSUE TITLE
The resource leak checker is taking a substantial amount of time on some NJR-1 projects.

### ISSUE THREAD

**Issue by iamsanjaymalakar** [typetools/checker-framework#6430]
I was running RLC on the benchmark projects of the NJR-1 dataset. It contains 293 java projects. Among them, two projects took more than `an hour` and still didn't finish running the RLC.

### Input
The NJR-1 dataset which is available at https://zenodo.org/records/4839913. Download the `njr-1_dataset.zip` file and extract it, where you will find all the projects.

The two projects that took an unsual amount of time are:
- url7a45f192ca_mfullen_ping_pong_net_tgz-pJ8-ping_pong_net_client_io_TestClientNonPPNServerJ8, `8226.978153467178`s, `6013` LoC
- url86119f019a_solzy_hive_parse_tgz-pJ8-org_apache_hadoop_hive_ql_parse_MainJ8, `1000.8362658023834`s, `53496` LoC
Here's the full list of the time taken by each project in the NJR-1 dataset (sorted desc on the timing, the timings are in seconds): 
[checkerframework_timings.csv](https://github.com/typetools/checker-framework/files/14103195/checkerframework_timings.csv)

### Commands
I invoked the CF using the Javac. Here's the abstracted command:
```
CF_BINARY = "../tools/checker-framework-3.42.0/checker/bin/javac"
CF_COMMAND = "-processor org.checkerframework.checker.resourceleak.ResourceLeakChecker -Adetailedmsgtext"
COMPILED_CLASSES_FOLDER="/ANY/FOLDER/TO/BE/USED/BY/CF"
lib_folder = "/THE/'lib'/FOLDER/INSIDE/EACH/BENCHMARK/PROJECT/WHICH/IS/PREBUILT/"
SRC_FILES="/A/TXT/FILE/CONTAINING/ALL/THE/JAVA/FILES/OF/THE/PROJECT/"
command = (CF_BINARY
        + " " + CF_COMMAND
        + " " + "-J-Xmx8G"
        + " " + "-d"
        + " " + COMPILED_CLASSES_FOLDER
        + " -cp " + lib_folder
        + " @" + SRC_FILES"
    )
```

### Output
Here's a screenshot of the timing of the two projects.
<img width="1247" alt="image" src="https://github.com/typetools/checker-framework/assets/32251702/9e3410b2-064c-46ff-9ce5-111f036d532e">

### Expectation
The expected behavoir is that it should not take one hour to run a project with only 6K LoC.

**Comment by msridhar** [comment]
Thanks @iamsanjaymalakar.  The first case is an infinite loop.  I've opened #6433 for that one.  Once we figure that one out I'll look at the second case.

**Comment by msridhar** [comment]
I've also opened an issue for the second case, #6435

**Comment by msridhar** [comment]
As #6433 is fixed I'm going to close this in favor of #6435 which covers the remaining issue
